### PR TITLE
[Flink 5404] Consolidate and update S3 documentation

### PR DESCRIPTION
**Pull request by medale**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [ ] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [ ] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [ ] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

**Comment by medale** [comment]
Error about RocksDB caused by: java.lang.UnsatisfiedLinkError: Native Library /tmp/librocksdbjni-linux64.so already loaded in another classloader at java.lang.ClassLoader.loadLibrary1(ClassLoader.java:1931). Since I only changed documentation this seems like an unrelated error with the overall build. How do I continue?

File: docs/dev/batch/connectors.md [hunk 1]
```diff
@@ -52,33 +52,13 @@ interface. There are Hadoop `FileSystem` implementations for
 
 In order to use a Hadoop file system with Flink, make sure that
 
-- the `flink-conf.yaml` has set the `fs.hdfs.hadoopconf` property set to the Hadoop configuration directory.
-- the Hadoop configuration (in that directory) has an entry for the required file system. Examples for S3 and Alluxio are shown below.
-- the required classes for using the file system are available in the `lib/` folder of the Flink installation (on all machines running Flink). If putting the files into the directory is not possible, Flink is also respecting the `HADOOP_CLASSPATH` environment variable to add Hadoop jar files to the classpath.
+- the `flink-conf.yaml` has set the `fs.hdfs.hadoopconf` property to the Hadoop configuration directory. For automated testing or running from an IDE the directory containing `flink-conf.yaml` can be set by defining the FLINK_CONF_DIR environment variable.
```
Comment (tzulitai) [inline docs/dev/batch/connectors.md]: Add \` around FLINK_CONF_DIR ==> `FLINK_CONF_DIR`

File: docs/setup/aws.md [hunk 2]
```diff
@@ -130,13 +134,13 @@ This registers `NativeS3FileSystem` as the default FileSystem for URIs with the
...
 
-You can specify the [Hadoop configuration]({{ site.baseurl }}/setup/config.html#hdfs) in various ways, for examples by configuring the path to the Hadoop configuration directory in `flink-conf.yaml`:
+You can specify the [Hadoop configuration]({{ site.baseurl }}/setup/config.html#hdfs) in various ways, for example by configuring the path to the Hadoop configuration directory in `flink-conf.yaml`:
 
 ```
 fs.hdfs.hadoopconf: /path/to/etc/hadoop
 ```
 
-This registers `path/to/etc/hadoop` as Hadoop's configuration directory with Flink.
+This registers `/path/to/etc/hadoop` as Hadoop's configuration directory with Flink. Flink will look for the "core-site.xml" and "hdfs-site.xml" files in the specified directory.
```
Comment (tzulitai) [inline docs/setup/aws.md]: Reference to core-site.xml and hdfs-site.xml here uses double apostrophes, while \` is used in other places.
Lets keep that styling consistent here :)

**Comment by tzulitai** [comment]
@medale The test failure is unrelated, it's ok to ignore that.

**Comment by medale** [comment]
@tzulitai  thank you for the feedback and catching those stylistic elements! I made the changes and recommitted.

**Comment by uce** [comment]
Thanks for the PR. Looks good to me. Merging this...

**Pull request closed without merging**